divvy/utils.py

```diff
@@ -296,6 +282,27 @@ def is_url(maybe_url):
     return urlparse(maybe_url).scheme != ""
 
 
+def parse_config_file(conf_file):
+    """
+    Parse a divvy configuration file.
+
+    :param str conf_file: path to divvy configuration file
+    :return Mapping: compute settings as declared in config file
+    """
+    with open(conf_file, 'r') as f:
+        _LOGGER.info("Loading divvy config file: %s", conf_file)
+        env_settings = yaml.load(f, yaml.SafeLoader)
+    _LOGGER.debug("Parsed environment settings: %s",
+                  str(env_settings))
+    # Any compute.submission_template variables should be made
+    # absolute, relative to current divvy configuration file.
+    if OLD_COMPUTE_KEY in env_settings:
+        warnings.warn("Divvy compute configuration '{}' section changed "
+                      "to '{}'".format(OLD_COMPUTE_KEY, NEW_COMPUTE_KEY),
+                      DeprecationWarning)
+        env_settings[NEW_COMPUTE_KEY] = env_settings[OLD_COMPUTE_KEY]
```

Comment:
It bothers me that this still allows subsequent mutations to have the keys get out of sync, but this is used in small enough fashion that I don't think it's a big deal. This sort of thing motivates https://github.com/pepkit/attmap/issues/20.
Same as https://github.com/pepkit/divvy/pull/28#discussion_r274067111